dvd/dvd_dump.ixx

```diff
@@ -505,6 +567,67 @@ export bool redumper_dump_dvd(Context &ctx, const Options &options, DumpMode dum
                     }
                 }
             }
+            else if(xgd_type != XGD_Type::NONE && !physical_structures.empty())
+            {
+                auto const &structure = physical_structures.front();
+
+                if(structure.size() < sizeof(CMD_ParameterListHeader) + sizeof(READ_DVD_STRUCTURE_LayerDescriptor))
+                    throw_line("invalid layer descriptor size (layer: {})", 0);
+
+                auto layer_descriptor = (READ_DVD_STRUCTURE_LayerDescriptor &)structure[sizeof(CMD_ParameterListHeader)];
+
+                uint32_t layer0_last = sign_extend<24>(endian_swap(layer_descriptor.layer0_end_sector));
```

Comment:
If we are doing sign extend that makes me think this data should be signed, should this be `int32_t` instead of unsigned variant? Same for `lba_first`